packages/react-server-cli/src/buildWebpackCompilers.js

```diff
@@ -0,0 +1,208 @@
+import webpack from "webpack"
+import path from "path"
+import mkdirp from "mkdirp"
+import fs from "fs"
+//import crypto from "crypto"
+import handleCompilationErrors from "./handleCompilationErrors";
+
+import normalizeRoutesPage from "./normalizeRoutesPage";
+
+
+// commented out to please eslint, but re-add if logging is needed in this file.
+// import {logging} from "react-server"
+// const logger = logging.getLogger(__LOGGER__);
+
+// compiles the routes file for browser clients using webpack.
+// returns a tuple of { compiler, serverRoutes }. compiler is a webpack compiler
+// that is ready to have run called, and serverRoutes is a promise that resolve to
+// a path to the transpiled server routes file path. The promise only resolves
+// once the compiler has been run. The file path returned from the promise
+// can be required and passed in to reactServer.middleware().
+// TODO: add options for sourcemaps.
+export default (opts = {}, webpackInfo) => {
+	const {
+		routes,
+		routesDir = ".",
+		hot,
+		longTermCaching,
+	} = opts;
+	const {
+		workingDirAbsolute,
+		outputDirAbsolute,
+		serverOutputDirAbsolute,
+		clientBootstrapFile,
+	} = webpackInfo.paths;
+
+	mkdirp.sync(workingDirAbsolute);
+	mkdirp.sync(outputDirAbsolute);
+	mkdirp.sync(serverOutputDirAbsolute);
+
+	writeClientBootstrapFile(clientBootstrapFile, opts);
+
+	// now rewrite the routes file out in a webpack-compatible way.
+	writeWebpackCompatibleRoutesFile(routes, routesDir, workingDirAbsolute, null, true);
```

Comment:
I know that this function was already called at this location before the PR, but why is it necessary?